src/main/java/com/linkedin/kmf/services/KeyedProduceService.java

```diff
@@ -0,0 +1,123 @@
+/**
+ * Copyright 2016 LinkedIn Corp. Licensed under the Apache License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.linkedin.kmf.services;
+
+import com.linkedin.kmf.common.Utils;
+import com.linkedin.kmf.producer.BaseProducerRecord;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.apache.kafka.common.utils.Utils.murmur2;
+
+
+public class KeyedProduceService extends ProduceService implements Service {
```

Comment:
This class does not have any documentation.  I know there are many Kafka monitor classes that do not have documentation, but really that's not a good idea.